torch/onnx/_internal/exporter.py

```diff
@@ -61,13 +69,46 @@ def __init__(
         *,
         opset_version: Optional[int] = None,
         dynamic_shapes: Optional[bool] = None,
+        op_level_debug: Optional[bool] = None,
         logger: Optional[logging.Logger] = None,
     ):
         self.opset_version = opset_version
         self.dynamic_shapes = dynamic_shapes
+        self.op_level_debug = op_level_debug
         self.logger = logger
 
 
+class ResolvedExportOptions:
+    @_beartype.beartype
+    def __init__(self, options: Optional[ExportOptions]):
+        if options is None:
+            options = ExportOptions()
+
+        T = TypeVar("T")
+
+        @_beartype.beartype
+        def resolve(value: Optional[T], fallback: Union[T, Callable[[], T]]) -> T:
+            if value is not None:
+                return value
+            if callable(fallback):
+                return fallback()
+            return fallback
+
+        self.opset_version = resolve(options.opset_version, _DEFAULT_OPSET_VERSION)
+        self.dynamic_shapes = resolve(options.dynamic_shapes, False)
+        self.op_level_debug = resolve(options.op_level_debug, False)
+        self.logger = resolve(
+            options.logger, lambda: logging.getLogger().getChild("torch.onnx")
+        )
+
```

Comment:
assert everything in `ExportOptions` are resolved?

```python
        # reference
        for key in dir(options):
            if key.startswith("_"):
                continue
            assert hasattr(self, key), f"Unresolved option '{key}'"
```

Reply:
I suggest put a comment on why if key.startswith("_"): decides resolved/unresolved.